lisa/tests/base.py

```diff
@@ -462,8 +472,37 @@ def test_noisy_tasks(self, noise_threshold_pct=None, noise_threshold_ms=None):
             threshold_s = min(threshold_s, noise_threshold_ms * 1e3)
 
         df = self.trace.analysis.tasks.df_tasks_runtime()
-        test_tasks = list(map(self.trace.get_task_pid, self.rtapp_profile.keys()))
-        df_noise = df[~df.index.isin(test_tasks + self.NOISE_IGNORED_PIDS)]
+
+        # We don't want to account the test tasks
+        ignored_pids = list(map(self.trace.get_task_pid, self.rtapp_profile.keys()))
+
+        def compute_duration_pct(row):
+            return row.runtime * 100 / self.trace.time_range
+
+        df["runtime_pct"] = df.apply(compute_duration_pct, axis=1)
+
+        # Figure out which PIDs to exclude from the thresholds
+        for key, threshold in self.NOISE_ACCOUNTING_THRESHOLDS.items():
+            # Find out which task(s) this threshold is about
+            if isinstance(key, int):
+                pids = [key]
+            elif isinstance(key, str):
+                pids = df[df.comm == key].index.values
+            elif isinstance(key, re._pattern_type):
```

Comment:
`_pattern_type` is a private attribute of `re` module, which seems undocumented and has been removed at some point between Python 3.5 and 3.7. It would probably be better to restrict to int and str, and just use `re.match()`. The `re` module has a compiled regex cache anyway, so if call that in a loop that should be fast enough (or we can call `.compile()` just before the loop if needed, but let's keep that local to this function.

Reply:
Argh, I didn't know about that removal - I thought that was the canonical way of checking this.

I'll simplify this as you've suggested.

Reply:
as a bonus that should make the Sphinx doc look better, since it displays the content of class attributes :)